src/gui_elements.py

```diff
@@ -0,0 +1,217 @@
+
+
+class Button(Clickable):
+
+    def __init__(self, onClick: Callable, *args,
+                 hitbox: Rect, inner_text: StylizedText,
+                 default_texture: pygame.color.Color | os.PathLike = pygame.color.Color(255, 255, 255),
+                 hover_texture: pygame.color.Color | os.PathLike = pygame.color.Color(160, 160, 160),
+                 click_texture: pygame.color.Color | os.PathLike = pygame.color.Color(64, 64, 64),
+                 border_radius: int = 0) -> None:
+
+        """
+        :param onClick (callback function):
+        :param *args (arguments for callback function):
+        :param  hitbox (rectangular):
+        :param inner_text: Текст на кнопке.
+        :param default_texture: Стандартная текстура кнопки.
+        :param hover_texture: Текстура при наведении курсора.
+        :param click_texture: Текстура при клике.
+        :param border_radius: Радиус округления.
+        """
+        super().__init__(onClick, *args, hitbox=hitbox)
+        self.inner_text: StylizedText = inner_text
+        self.default_texture: pygame.color.Color | os.PathLike = default_texture
+        self.hover_texture: pygame.color.Color | os.PathLike = hover_texture
+        self.click_texture: pygame.color.Color | os.PathLike = click_texture
+        self.button_texture: pygame.color.Color | os.PathLike = self.default_texture
+        self.border_radius: int = border_radius
+        self._image_cache: dict = {}
+
+    def hover_click(self, event: pygame.event) -> None:
+        """
+        Красит кнопку в нужный цвет.
+        :param event: Действия пользователя
+        """
+        collide = super().check_collision()
+        if collide:
+            self.button_texture = self.hover_texture
+            if event.type == pygame.MOUSEBUTTONDOWN:
+                self.button_texture = self.click_texture
+            elif event.type == pygame.MOUSEBUTTONUP:
+                self.button_texture = self.hover_texture
+
+        else:
+            self.button_texture = self.default_texture
+
+    def render(self, screen: pygame.surface) -> None:
+        """
+        Выводит кнопку на экран.
+        :param screen: Объект дисплея для обновления содержимого.
+        """
+        self.inner_text.position = center_relative_to(self.inner_text.position, self.hitbox)
+        if isinstance(self.button_texture, pygame.color.Color):
+            pygame.draw.rect(screen, self.button_texture, self.hitbox, 0,
+                             border_radius=self.border_radius)
+        elif isinstance(self.button_texture, os.PathLike):
+            # Кэшировние изображений, которые уже были зарендерины.
+            if self.button_texture not in self._image_cache:
+                img = pygame.image.load(self.button_texture).convert_alpha()
+                if img.get_size() != (self.hitbox[2], self.hitbox[3]):
+                    img = pygame.transform.smoothscale(img, (self.hitbox[2], self.hitbox[3]))
```

Comment:
Для корректного отображения желательно еще использовать convert_alpha
```suggestion
                    img = pygame.transform.smoothscale(img.convert_alpha(), (self.hitbox[2], self.hitbox[3]))
```